test/specs/testCase4.js

```diff
@@ -0,0 +1,16 @@
+import { assert } from 'chai';
+import ProductPage from '../../pageobject/productpage/ProductPage.js';
+import HomePage from '../../pageobject/homepage/HomePage.js';
+const homepage = new HomePage();
+const productpage = new ProductPage();
+
+describe('Test case 4', () => {
+    it('Check login', async () => {
+        await homepage.openHomePage();
+        await homepage.logIn('performance_glitch_user','secret_sauce');
+        await browser.waitUntil(async () =>{
+            return productpage.isProductTitleDisplayed();
+        });
```

Comment:
I'd suggest creating some utility class for waitings
something like:

```
class Waits {
  async waitUntil (condition, {
    timeout = Timeouts.DEFAULT_WAIT_TIMEOUT,
    interval = Timeouts.DEFAULT_WAIT_INTERVAL,
    timeoutMsg = `Condition not met in ${timeout / 1000} seconds`} = {}) {
  // ... your code here
  }
}
```

and using it in such cases